src/highdicom/sr/templates.py

```diff
@@ -3424,14 +3683,21 @@ def __init__(
                     ),
                     relationship_type=RelationshipTypeValues.CONTAINS
                 )
-                group_item.ContentSequence = descriptor_items
-                # The Image Library Entry template contains the individual
-                # Image Library Entry Descriptors content items.
-                if not isinstance(descriptor_items,
-                                  ImageLibraryEntryDescriptors):
-                    raise TypeError(
-                        'Image library group items must have type '
-                        '"ImageLibraryEntry".'
-                    )
+                group_item.ContentSequence = ContentSequence()
+                group_item.ContentSequence.extend(library_item_entry)
+
+                image = ImageContentItem(
```

Comment:
I would append the name with `_item` for consistency

```suggestion
                image_item = ImageContentItem(
```